agent/consul/acl_endpoint.go

```diff
@@ -915,13 +915,13 @@ func (a *ACL) PolicySet(args *structs.ACLPolicySetRequest, reply *structs.ACLPol
 			return fmt.Errorf("Invalid Policy: A policy with name %q already exists", policy.Name)
 		}
 
-		if policy.ID == structs.ACLPolicyGlobalManagementID {
+		if builtinPolicy, ok := structs.ACLBuiltinPolicies[policy.ID]; ok {
 			if policy.Datacenters != nil || len(policy.Datacenters) > 0 {
-				return fmt.Errorf("Changing the Datacenters of the builtin global-management policy is not permitted")
+				return fmt.Errorf("Changing the Datacenters of the builtin %s policy is not permitted", builtinPolicy.Name)
```

Comment:
I think this error message is fine for the existing global management policy but it will stutter for new global read only policy:
```
Changing the Datacenters of the builtin builtin/global-read-only policy is not permitted
```
This is part of a the reason I'm wondering if we can name them in the same way (previous comment).

Reply:
We can do this instead
```suggestion
				return fmt.Errorf("Changing the Datacenters of the %s policy is not permitted", builtinPolicy.Name)
```

Reply:
That looks good to me.